snips_nlu/intent_parser/lookup_intent_parser.py

```diff
@@ -0,0 +1,403 @@
+
+    @fitted_required
+    def get_intents(self, text):
+        """Returns the list of intents ordered by decreasing probability
+
+        The length of the returned list is exactly the number of intents in the
+        dataset + 1 for the None intent
+        """
+        intents = []
+        res = self.parse(text)
+        try:
+            matched_intent = res[RES_INTENT]
+            intent_name = matched_intent[RES_INTENT_NAME]
+            intents.append(matched_intent)
+            others = [x for x in self.intents_names if x != intent_name]
+        except TypeError:
+            others = [x for x in self.intents_names]
+
+        for intent in others:
+            intents.append(intent_classification_result(intent, 0.0))
+
+        intents.append(intent_classification_result(None, 0.0))
+
+        return intents
+
+    @fitted_required
+    def get_slots(self, text, intent):
+        """Extracts slots from a text input, with the knowledge of the intent
+
+        Args:
+            text (str): input
+            intent (str): the intent which the input corresponds to
+
+        Returns:
+            list: the list of extracted slots
+
+        Raises:
+            IntentNotFoundError: When the intent was not part of the training
+                data
+        """
+        if intent is None:
+            return []
+
+        if intent not in self.intents_names:
+            raise IntentNotFoundError(intent)
+
+        slots = self.parse(text, intents=[intent])[RES_SLOTS]
+        if slots is None:
+            slots = []
+        return slots
+
+    def get_intent_id(self, intent_name):
+        intent_id = self._intents_mapping.get(intent_name)
+        if intent_id is None:
+            intent_id = len(self.intents_names)
+            self.intents_names.append(intent_name)
+            self._slots_mapping[intent_name] = intent_id
+
+        return intent_id
+
+    def get_slot_id(self, slot_name):
+        slot_id = self._slots_mapping.get(slot_name)
+        if slot_id is None:
+            slot_id = len(self.slots_names)
+            self.slots_names.append(slot_name)
+            self._slots_mapping[slot_name] = slot_id
+
+        return slot_id
+
+    def _preprocess_text(self, txt):
+        """Replaces stop words and characters that are tokenized out by
+            whitespaces"""
+        tokens = tokenize(txt, self.language)
+        current_idx = 0
+        cleaned_string = ""
+        for token in tokens:
+            if self.stop_words and normalize_token(token) in self.stop_words:
+                token.value = "".join(" " for _ in range(len(token.value)))
+            prefix_length = token.start - current_idx
+            cleaned_string += "".join((" " for _ in range(prefix_length)))
+            cleaned_string += token.value
+            current_idx = token.end
+        suffix_length = len(txt) - current_idx
+        cleaned_string += "".join((" " for _ in range(suffix_length)))
+        return cleaned_string
```

Comment:
This function can be simplified: as we don't need the ranges, you can just tokenize, lowercase, filter stop words, and then join with `" "`.